File: OrangeJuice.Server.Core/Data/Configuration/AzureDbConfiguration.cs
using System.Data.Entity;
using System.Data.Entity.SqlServer;

namespace OrangeJuice.Server.Data.Configuration
{
	public sealed class AzureDbConfiguration : DbConfiguration
	{
		public AzureDbConfiguration()
		{
			SetExecutionStrategy("System.Data.SqlClient", () => new SqlAzureExecutionStrategy());
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Data/AzureDbConfiguration.fs
namespace OrangeJuice.Server.FSharp.Data.Configuration

open System
open System.Data.Entity
open System.Data.Entity.Infrastructure
open System.Data.Entity.SqlServer

type AzureDbConfiguration() =
    inherit DbConfiguration()
    do
        let f = fun () -> new SqlAzureExecutionStrategy() :> IDbExecutionStrategy
        base.SetExecutionStrategy("System.Data.SqlClient", new Func<IDbExecutionStrategy>(f))
